libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/LegWaypointFactory.kt

```diff
@@ -9,6 +10,7 @@ object LegWaypointFactory {
         location: Point,
         name: String,
         target: Point?,
-        @LegWaypoint.Type type: String
-    ): LegWaypoint = LegWaypoint(location, name, target, type)
+        @LegWaypoint.Type type: String,
+        metadata: Map<String, JsonElement>?,
```

Comment:
Can we introduce a wrapper for this map that provides some predefined fields?

For example,
```
@ExperimentalPreviewMapboxNavigationAPI
class WaypointMetadata(val rawMetadata: Map<String, JsonElement>) {
    val chargeTime: String? = rawMetadata["charge_time"]?.getAsString()
    ...
}
```

Reply:
We don't do that because it's in beta on server side. That's why we don't have any type-safe bindings, in mapbox-java as well, that's by design.